model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/user/entity/JpaUserEntity.java

```diff
@@ -62,11 +62,7 @@
  * @author <a href="mailto:sguilhen@redhat.com">Stefan Guilhen</a>
  */
 @Entity
-@Table(name = "kc_user",
-        uniqueConstraints = {
-                @UniqueConstraint(columnNames = {"realmId", "username"}),
-                @UniqueConstraint(columnNames = {"realmId", "emailConstraint"})
-        })
```

Comment:
I've noticed there are no constraints in the db, therefore this is removed from here. I was wondering whether we should add the constrains instead of removing it from here.

Reply:
I this we should add the constraints for
`@UniqueConstraint(columnNames = {"realmId", "username", "usernameWithCase"})` - `username` is present always, and if the same, it could only be due to difference in `usernameWithCase`. WDYT?

Reply:
It makes sense to me. Would you prefer do it within this issue or should I create a new one to deal with the constraints?

Reply:
I would prefer a new issue.

Reply:
https://github.com/keycloak/keycloak/issues/14797